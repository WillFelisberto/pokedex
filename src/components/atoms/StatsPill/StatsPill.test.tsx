import { render, screen } from '@/tests/test-utils';

import { StatsPill } from '.';

describe('<StatsPill />', () => {
  it('should render the component', () => {
    const { container } = render(<StatsPill title="StatsPill" />);

    expect(screen.getByText(/StatsPill/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should have the correct styles', () => {
    render(<StatsPill title="StatsPill" />);

    const statsPill = screen.getByText(/StatsPill/i);
    expect(statsPill).toHaveClass('shadow-lg');
    expect(statsPill).toHaveClass('font-semibold');
    expect(statsPill).toHaveClass('uppercase');
    expect(statsPill).toHaveClass('text-center');
    expect(statsPill).toHaveClass('text-gray-700');
    expect(statsPill).toHaveClass('bg-white');
    expect(statsPill).toHaveClass('rounded-xl');
    expect(statsPill).toHaveClass('py-2');
    expect(statsPill).toHaveClass('px-5');
    expect(statsPill).toHaveClass('hover:bg-gray-100');
  });
});
