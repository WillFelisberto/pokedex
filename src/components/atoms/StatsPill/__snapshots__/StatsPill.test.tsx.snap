// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatsPill /> should render the component 1`] = `
<div>
  <p
    class="text-xs w-fit shadow-lg font-semibold uppercase text-center text-gray-700 bg-white rounded-xl py-2 px-5 hover:bg-gray-100"
  >
    StatsPill
  </p>
</div>
`;
